lua/refactoring/tasks/adjust_cursor.lua

```diff
@@ -37,13 +37,15 @@ local function adjust_cursor(refactor)
     local add_rows = get_rows(cursor)
     local r, _ = cursor:to_vim_win()
     local result_row = r + add_rows
-    vim.schedule(function()
-        vim.api.nvim_win_set_cursor(win, {
-            result_row,
-            refactor.highlight_start_col - 1,
-        })
-    end)
+
     reset()
+    utils.wait_frame()
+
+    local _, col = cursor:to_vim()
+    vim.api.nvim_win_set_cursor(win, {
+        result_row,
+        col,
```

Comment:
gives best refactoring column I think.

Reply:
While is may be better for `extract`, I think it gives a worse experience than those for `extract_var` on `inline_var`. I think instead we should do something like this: 
```lua
local col
if refactor.highlight_start_col ~= nil then
    col = refactor.highlight_start_col
else
    _, col = cursor:to_vim()
end
```
Then remove `highlight_start_col` from `106` so that it has better cursor while the positioning. I was gonna work on making extract func  column movement (and maybe debug if it has the same issue) better by searching for the first nonwhite space character in the column we are setting the cursor to. Let me know your thoughts. 

Other than this the debug stuff looks good. Once merged I can start trying to make some tests around it.

Reply:
If you want to merge as is I change to what I mean above as well, whatever is easier for you.

Reply:
@teddylear 

This is what I am thinking.  We drop the formatting motion.  Let the LSP keep the cursor in the correct position.  I think this would help the most.

We may have to do some cursor movement ourselves, but until then we should just use neoformat.  I think that would be the biggest help.